esphome/components/kamstrup_kmp/kamstrup_kmp.cpp

```diff
@@ -0,0 +1,315 @@
+#include "kamstrup_kmp.h"
+
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace kamstrup_kmp {
+
+static const char *const TAG = "kamstrup_kmp";
+
+void KamstrupKMPComponent::setup() {}
+
+void KamstrupKMPComponent::dump_config() {
+  ESP_LOGCONFIG(TAG, "kamstrup_kmp:");
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "Communication with Kamstrup meter failed!");
+  }
+  LOG_UPDATE_INTERVAL(this);
+
+  if (this->heat_energy_sensor_ != nullptr) {
+    LOG_SENSOR("  ", "Heat Energy", this->heat_energy_sensor_);
+  }
```

Comment:
There is a nullptr check inside this macro already so its not needed here
```suggestion
  LOG_SENSOR("  ", "Heat Energy", this->heat_energy_sensor_);
```